File: OOP_final_project_python_version/escola.py
from pessoa_juridica import PessoaJuridica

class Escola(PessoaJuridica):
    def __init__(self, nome, nome_fantasia, num_funcionario, nome_oficial, cpf_cnpj, endereco, telefone):
        super().__init__(nome_fantasia, num_funcionario, nome_oficial, cpf_cnpj, endereco, telefone)
        self.__nome = nome
        self.__alunos = []

    def apresentar_dados(self):
        super().apresentar_dados()
        print("Nome: " + self.nome)
        print("CNPJ: " + self.cpf_cnpj)
        print("Telefone: " + self.telefone)
        print("Endereço: " + str(self.endereco))
        print("Alunos Matriculados: " + self.exibir_alunos())

    @property
    def nome(self):
        return self.__nome

    @nome.setter
    def nome(self, nome):
        self.__nome = nome

    def matricular_aluno(self, aluno):
        self.__alunos.append(aluno)
        return "Aluno matriculado com sucesso"

    def exibir_alunos(self):
        t = " "
        for aluno in self.__alunos:
            t += " " + self.listar_alunos(aluno)
        return t

    def listar_alunos(self, aluno):
        return "\nMatrícula: " + str(aluno.matricula) + " | Nome: " + aluno.nome_oficial + " | Série: " + aluno.serie

    # Demais getters e setters herdados de PessoaJuridica podem ser mantidos aqui

File: OOP_final_project_JAVA_version/src/exercicio_incremental_parte2/Escola.java
package exercicio_incremental_parte2;

import java.util.ArrayList;
import java.util.Iterator;

public class Escola extends PessoaJuridica {
    private String nome;
    private ArrayList<Aluno> aluno;

    public Escola(String nome, String nome_fantasia, int num_funcionario, String nome_oficial, String cpf_cnpj, Endereco endereco, String telefone) {
        super(nome_fantasia, num_funcionario, nome_oficial, cpf_cnpj, endereco, telefone);
        this.nome = nome;
        this.aluno = new ArrayList<>();
        
    }

    @Override
public void apresentarDados() {
    super.apresentarDados(); 
    System.out.println("Nome: " + getNome());
    System.out.println("CNPJ: " + getCpf_cnpj());
    System.out.println("Telefone: " + getTelefone());
    System.out.println("Endereço: " + getEndereco());
    System.out.println("Alunos Matriculados: " + exibirAlunos());
}

 @Override
    public String verificarTipo() {
        String tipoSuperClasse = super.verificarTipo();
        return tipoSuperClasse + " Escola";
    }
    
    /*
    public Escola(String nome, String cnpj, String telefone) {
        this.nome = nome;
        this.cnpj = cnpj;
        this.telefone = telefone;
        this.aluno = new ArrayList<>();
    }
*/
    
    public String matricularAluno(Aluno aluno){
        this.aluno.add(aluno);
        return "Aluno matriculado com sucesso";
    }
    
    public String exibirAlunos(){
        String t = " ";
        for(Iterator<Aluno>it = aluno.iterator();it.hasNext();){
            Aluno x = it.next();
            t = t + " " + listarAlunos(x);
        }return t;
    }
    
    public String listarAlunos(Aluno aluno){
        return "\nMatricula: " + aluno.getMatricula()+" | nome social: "+ aluno.getNome_oficial()+" | serie: "+aluno.getSerie();
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    
}
